### PR TITLE
Epic popup: updates and fixes

### PR DESCRIPTION
**Pull request by dshovchko**
In the scope of this PR:
- update static methods on esl-tooltip to have these methods out of the box in the inherited classes
- fix focus out behavior after SHIFT+TAB keypress on esl-tooltip
- add afterOnShow() and beforeOnHide() methods to better flexibility and clear understanding

### CODE REVIEW

**Review by ala-n — APPROVED** [review]

**Review by julia-murashko — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

File: src/modules/esl-popup/core/esl-popup.ts [hunk 1]
```diff
@@ -163,6 +158,7 @@ export class ESLPopup extends ESLToggleable {
    * Removes CSS classes and update a11y by default.
```
Comment (fshovchko) [inline src/modules/esl-popup/core/esl-popup.ts:158]: Comment needs small correction: "Removes CSS classes and updates a11y by default."

File: src/modules/esl-tooltip/core/esl-tooltip.ts [hunk 2]
```diff
@@ -127,6 +128,21 @@ export class ESLTooltip extends ESLPopup {
     }
   }
 
+  /**
+   * Actions to execute after show popup.
```
Comment (fshovchko) [inline src/modules/esl-tooltip/core/esl-tooltip.ts:132]: Comment sounds more correct like this: "Actions to execute after popup show."

File: src/modules/esl-tooltip/core/esl-tooltip.ts [hunk 3]
```diff
@@ -127,6 +128,21 @@ export class ESLTooltip extends ESLPopup {
...
+   */
+  protected afterOnShow(): void {
+    super.afterOnShow();
+    this.focus({preventScroll: true});
+  }
+
+  /**
+   * Actions to execute before hide popup.
```
Comment (fshovchko) [inline src/modules/esl-tooltip/core/esl-tooltip.ts:140]: Comment sounds more correct like this: "Actions to execute after popup hide."

Comment (fshovchko) [reply src/modules/esl-tooltip/core/esl-tooltip.ts:140]: "Actions to execute after show of popup" sounds alright aswell

File: src/modules/esl-tooltip/core/esl-tooltip.ts (hunk 2 above)
Comment (fshovchko) [reply src/modules/esl-tooltip/core/esl-tooltip.ts:132]: Actions to execute after show of popup" sounds alright aswell

File: src/modules/esl-tooltip/core/esl-tooltip.ts [hunk 4]
```diff
@@ -110,15 +116,10 @@ export class ESLTooltip extends ESLPopup {
     document.body.appendChild(this);
     super.onShow(params);
     this._updateActivatorState(true);
-
-    setTimeout(() => {
-      this.focus();
-    });
   }
 
   /** Actions to execute on hide Tooltip. */
```
Comment (fshovchko) [inline src/modules/esl-tooltip/core/esl-tooltip.ts:121]: Comment sounds more correct like this: "Actions to execute after Tooltip hide." or "Actions to execute after hide of Tooltip"

File: src/modules/esl-popup/core/esl-popup.ts [hunk 5]
```diff
@@ -174,6 +170,21 @@ export class ESLPopup extends ESLToggleable {
     memoize.clear(this, '_offsetArrowRatio');
   }
 
+  /**
+   * Actions to execute after show popup.
```
Comment (fshovchko) [inline src/modules/esl-popup/core/esl-popup.ts:174]: Comment sounds more correct like this: "Actions to execute after popup show." or "Actions to execute after show of popup"

File: src/modules/esl-popup/core/esl-popup.ts [hunk 6]
```diff
@@ -174,6 +170,21 @@ export class ESLPopup extends ESLToggleable {
...
+   */
+  protected afterOnShow(): void {
+    this._updatePosition();
+    this.style.visibility = 'visible';
+    this.activator && this._addActivatorObserver(this.activator);
+    this._startUpdateLoop();
+  }
+
+  /**
+   * Actions to execute before hide popup.
```
Comment (fshovchko) [inline src/modules/esl-popup/core/esl-popup.ts:184]: Comment sounds more correct like this: "Actions to execute after popup hide" or "Actions to execute after hide of popup"

**Review by Anna-MariiaPetryk — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Pull request merged**